python-macos/config.sh

```diff
@@ -21,37 +21,94 @@ set -e
 
 function build_wheel {
     echo `pwd`
+    export PATH="/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH"
     echo CFLAGS=${CFLAGS}
     echo CXXFLAGS=${CXXFLAGS}
     echo LDFLAGS=${LDFLAGS}
     pushd $1
 
-	wget --no-check-certificate http://downloads.sourceforge.net/project/boost/boost/1.60.0/boost_1_60_0.tar.gz -O boost_1_60_0.tar.gz
-	tar xf boost_1_60_0.tar.gz
-    pushd boost_1_60_0
+    boost_version="1.65.1"
+    boost_directory_name="boost_${boost_version//\./_}"
+    boost_tarball_name="${boost_directory_name}.tar.gz"
+    wget --no-check-certificate \
+        http://downloads.sourceforge.net/project/boost/boost/"${boost_version}"/"${boost_tarball_name}" \
+        -O "${boost_tarball_name}"
+    tar xf "${boost_tarball_name}"
+
+    arrow_boost="$PWD/arrow_boost"
+    arrow_boost_dist="$PWD/arrow_boost_dist"
+    mkdir "$arrow_boost" "$arrow_boost_dist"
+    pushd "${boost_directory_name}"
+
+    # Arrow is 64-bit-only at the moment
+    export CFLAGS="-fPIC -arch x86_64 ${CFLAGS//"-arch i386"/}"
+    export CXXFLAGS="-fPIC -arch x86_64 ${CXXFLAGS//"-arch i386"} -std=c++11"
+
+    ./bootstrap.sh
+    ./b2 tools/bcp
+    ./dist/bin/bcp --namespace=arrow_boost --namespace-alias \
+        filesystem date_time system regex build algorithm locale format "$arrow_boost"
+
+    popd
+    pushd "$arrow_boost"
     ./bootstrap.sh
-    ./bjam "cxxflags=-fPIC ${CFLAGS}" cflags="-fPIC ${CXXFLAGS}" --prefix=/usr/local --with-filesystem --with-date_time --with-system --with-regex install
+    ./bjam cxxflags="${CXXFLAGS}" \
+        linkflags="-std=c++11" \
+        cflags="${CFLAGS}" \
+        variant=release \
+        link=shared \
+        --prefix="$arrow_boost_dist" \
+        --with-filesystem --with-date_time --with-system --with-regex \
+        install
     popd
 
-    # Arrow is 64bit-only at the moment
-    export CFLAGS="-arch x86_64"
-    export CXXFLAGS="-arch x86_64"
-    export ARROW_HOME=/usr/local/
-    export PARQUET_HOME=/usr/local/
-    pip install cython==0.25.2 numpy==${NP_TEST_DEP}
+    export THRIFT_HOME=/usr
+    export THRIFT_VERSION=0.11.0
+    wget http://archive.apache.org/dist/thrift/${THRIFT_VERSION}/thrift-${THRIFT_VERSION}.tar.gz
+    tar xf thrift-${THRIFT_VERSION}.tar.gz
+    pushd thrift-${THRIFT_VERSION}
+    mkdir build-tmp
+    pushd build-tmp
+    cmake -DCMAKE_BUILD_TYPE=release \
+        "-DCMAKE_CXX_FLAGS=-fPIC" \
+        "-DCMAKE_C_FLAGS=-fPIC" \
+        "-DCMAKE_INSTALL_PREFIX=${THRIFT_HOME}" \
+        "-DCMAKE_INSTALL_RPATH=${THRIFT_HOME}/lib" \
+        "-DBUILD_SHARED_LIBS=OFF" \
+        "-DBUILD_TESTING=OFF" \
+        "-DWITH_QT4=OFF" \
+        "-DWITH_C_GLIB=OFF" \
+        "-DWITH_JAVA=OFF" \
+        "-DWITH_PYTHON=OFF" \
+        "-DWITH_CPP=ON" \
+        "-DWITH_STATIC_LIB=ON" \
+        "-DWITH_LIBEVENT=OFF" \
+        -DBoost_NAMESPACE=arrow_boost \
+        -DBOOST_ROOT="$arrow_boost_dist" \
+        ..
+    make install -j5
```

Comment:
Looks like this is failing because we are unable to copy a file:

```
-- Install configuration: "release"

-- Installing: /usr/bin/thrift

CMake Error at compiler/cpp/cmake_install.cmake:36 (file):

  file INSTALL cannot copy file

  "/Users/travis/build/cpcloud/arrow-dist/arrow/thrift-0.11.0/build-tmp/compiler/cpp/bin/thrift"

  to "/usr/bin/thrift".

Call Stack (most recent call first):

  cmake_install.cmake:37 (include)
```

@xhochy Do we need to use the `sudo` hammer here?

Reply:
no, this should go into `/usr/local`, fixing.

Reply:
great thx